tiledb/sm/subarray/subarray.cc

```diff
@@ -114,17 +114,33 @@ Subarray::Subarray(
     shared_ptr<Logger> logger,
     const bool coalesce_ranges,
     StorageManager* storage_manager)
+    : Subarray(
+          array->opened_array(),
+          layout,
+          parent_stats,
+          logger,
+          coalesce_ranges,
+          storage_manager) {
+}
+
+Subarray::Subarray(
+    const shared_ptr<OpenedArray> opened_array,
+    const Layout layout,
+    Stats* const parent_stats,
+    shared_ptr<Logger> logger,
+    const bool coalesce_ranges,
+    StorageManager* storage_manager)
     : stats_(
           parent_stats ? parent_stats->create_child("Subarray") :
           storage_manager ?
                          storage_manager->stats()->create_child("subSubarray") :
                          nullptr)
     , logger_(logger->clone("Subarray", ++logger_id_))
-    , array_(array)
+    , array_(opened_array)
```

Comment:
I know it'll increase the line count, but this strikes me as a bit sketch that we're changing the type here and not the array name. I'd wager that if anyone sees an internal variable named `array` they'll assume it has `tiledb::sm::Array` as its type. Suddenly having to verify which type an `array` variable is seems like it'll be a greater long term burden than just updating the variable name in this PR.

Reply:
As discussed, one we make array C.41 compliant, this will switch back to an array object. More, with our current plan for views, we are likely to remove array completely from here and replace with view.